python/cudaq/kernel/kernel_decorator.py

```diff
@@ -142,8 +156,28 @@ def __call__(self, *args):
         kernel AST to MLIR will occur here if it has not already occurred. 
         """
 
-        if self.module == None:
-            self.compile()
+        # Before we can execute, we need to make sure
+        # variables from the parent frame that we captured
+        # have not changed. If they have changed, we need to
+        # recompile with the new values.
```

Comment:
Just to double check, does this section do a deep check of the values? I.e., if a single list element changes, will it detect that and recompile?

Reply:
The check is that the old value is `==` to the new value. I think that will evaluate to `False` for lists with different elements.